### PR TITLE
Kics recommendations

### PR DESCRIPTION
**Pull request by vara-bonthu**
*Issue #, if available:*

*Description of changes:*
- This PR addresses majority of the KICS recommended security alerts

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: modules/kubernetes-addons/variables.tf [hunk 1]
```diff
@@ -220,6 +224,23 @@ variable "aws_for_fluentbit_irsa_policies" {
   default     = []
 }
 
+variable "aws_for_fluentbit_cw_log_group_name" {
+  type        = string
+  description = "FluentBit CloudWatch Log group name"
+  default     = null
```
Comment (kcoleman731) [inline modules/kubernetes-addons/variables.tf:230]: lets provide a default?

File: modules/kubernetes-addons/aws-for-fluentbit/variables.tf [hunk 2]
```diff
@@ -27,6 +27,24 @@ variable "eks_cluster_id" {
   description = "EKS cluster Id"
 }
 
+variable "cw_log_group_name" {
+  type        = string
+  description = "FluentBit CloudWatch Log group name"
+  default     = null
```
Comment (kcoleman731) [inline modules/kubernetes-addons/aws-for-fluentbit/variables.tf:33]: default

File: modules/kubernetes-addons/aws-for-fluentbit/main.tf [hunk 3]
```diff
@@ -65,18 +60,33 @@ resource "helm_release" "aws_for_fluent_bit" {
...
+  source                     = "../../../modules/irsa"
+  eks_cluster_id             = var.eks_cluster_id
+  kubernetes_namespace       = local.helm_config["namespace"]
+  kubernetes_service_account = local.service_account_name
+  irsa_iam_policies          = concat([aws_iam_policy.aws_for_fluent_bit.arn], var.irsa_policies)
+  tags                       = var.tags
+}
+
+module "kms" {
+  source      = "../../../modules/aws-kms"
```
Comment (kcoleman731) [inline modules/kubernetes-addons/aws-for-fluentbit/main.tf:87]: do you need conditional here? In case customer supplies cw_log_group_kms_key_arn?

**Review by kcoleman731 — COMMENTED** [review]
Big ol PR. Looks good.

File: modules/kubernetes-addons/variables.tf (hunk 1 above)
Comment (vara-bonthu) [reply modules/kubernetes-addons/variables.tf:230]: We have an option in locals to create a opinionated log group if the users doesn't provide any

`log_group_name       = var.cw_log_group_name == null ? "/${var.eks_cluster_id}/worker-fluentbit-logs" : var.cw_log_group_name`

File: modules/kubernetes-addons/aws-for-fluentbit/variables.tf (hunk 2 above)
Comment (vara-bonthu) [reply modules/kubernetes-addons/aws-for-fluentbit/variables.tf:33]: yes, default is null. see above response

File: modules/kubernetes-addons/aws-for-fluentbit/main.tf (hunk 3 above)
Comment (vara-bonthu) [reply modules/kubernetes-addons/aws-for-fluentbit/main.tf:87]: Good catch! I will update this now

**Review by askulkarni2 — APPROVED** [review]
Looks good.

**Review by kcoleman731 — APPROVED** [review]
LGTM

**Pull request merged**